### ISSUE TITLE
Double colon route breaks default fallback when there is another route with a common prefix.

### ISSUE THREAD

**Issue by emfomy** [delvedor/find-my-way#219]
I register two routes:

- `GET /hello`
- `GET /hello:world`

Both routes work fine. However, the server freeze when sending request to `/hello:worl`, `/hello:world1`, `/hello:bar`, `/hello123` (i.e. and route start with `/hello`).

Without `GET /hello` registered, these routes can fall back to 404 properly.

### Code

```javascript
const http = require("http");
const router = require("find-my-way")();

router.on("GET", "/hello", (req, res, params) => {
  res.end("hello");
});

router.on("GET", "/hello::world", (req, res, params) => {
  res.end("hello world");
});

const server = http.createServer((req, res) => {
  router.lookup(req, res);
});

server.listen(3000, (err) => {
  if (err) throw err;
  console.log("Server listening on: http://localhost:3000");
});
```

### Related Issue
- fastify/fastify#3550

**Comment by emfomy** [comment]
According to @climba03003 (https://github.com/fastify/fastify/issues/3550#issuecomment-997624300), it is related to the infinite loop in `.find` method.

https://github.com/delvedor/find-my-way/blob/695654cdd452d2993ada90cc87bdf8d695aee076/index.js#L497-L505

**Comment by emfomy** [comment]
Fixed in [v5.1.1](https://github.com/delvedor/find-my-way/releases/tag/v5.1.1).
